src/controllers/authentication/user.js

```diff
@@ -9,52 +12,57 @@ import { responseFormat, errorResponseFormat } from '../../utils/index';
 };
 
 export const createUser = async (req, res) => {
   try {
     const { body } = req;
-    const user = await User.create({ ...body });
+    const verificationToken = uuid();
+    const user = await User.create({ ...body, verificationToken });
     const { id, username, email, role, fullName, bio } = user;
 
+    let url = `http://${process.env.HOST}:${process.env.PORT}/api/v1/verify`;
+
+    if (process.env.NODE_ENV === 'production') { url = `https://${process.env.HOST}:${process.env.PORT}/api/v1/verify`; }
+
+    const verificationUrl = `${url}/${id}/${verificationToken}`;
+
+    const subject = 'Welcome to Authors\' Haven';
+
+    const message = `
+      <div>
+        <p>Hi ${fullName},</p>
+        <p>Welcome to Authors Haven, a place to be inspired! Your account was successfully created.</p>
+        <p>Please click this <a href=${verificationUrl}>link</a> to confirm your account.</p>
+      </div>`;
+
     if (user) {
-      return res.status(201).json(responseFormat({
-        status: 'success',
-        data: {
-          token: Auth.generateToken({
-            id, fullName, bio, email, username, role
-          })
-        },
+      const verificationPayload = { id, fullName, email, subject, message };
+      sendMail(verificationPayload);
+      return res.status(201).json(responseFormat({ status: 'success',
+        data: { token: Auth.generateToken({ id, fullName, bio, email, username, role }) },
       }));
     }
   } catch (error) {
     if (error.name === 'SequelizeUniqueConstraintError') {
-      if (error.fields.email) {
-        return res.status(409).json(errorResponseFormat({
-          message: 'This Email Already Exist',
-        }));
-      }
-
-      if (error.fields.username) {
-        return res.status(409).json(errorResponseFormat({
-          message: 'This Username Already Exist',
-        }));
-      }
+      if (error.fields.email) { return res.status(409).json(errorResponseFormat({ status: 'fail', message: 'This Email Already Exist' })); }
+
+      if (error.fields.username) { return res.status(409).json(errorResponseFormat({ status: 'fail', message: 'This Username Already Exist' })); }
     }
+    return res.status(500).json(errorResponseFormat({ message: 'Something Went Wrong', }));
+  }
+};
 
-    return res.status(500).json(errorResponseFormat({
-      message: 'Something Went Wrong',
-    }));
+export const verifyUser = async (req, res) => {
+  try {
+    const { id, verificationToken } = req.params;
+    let user = await User.findOne({ where: { id, verificationToken } });
+    if (!user) { return res.status(403).json(errorResponseFormat({ status: 'fail', message: 'Invalid token supplied, kindly reauthenticate!', })); }
+    user = await user.update({ verified: true, verificationToken: null });
+    if (user) { return res.status(202).json(responseFormat({ status: 'success', data: 'Your account was successfully verified!', })); }
```

Comment:
Think of line 62 and 64, it is difficult to reason. Let us improve on the names

Reply:
I have made it more readable. Thanks